internal/downlink/data/data.go

```diff
@@ -387,14 +391,177 @@ func setTXParameters(ctx *dataContext) error {
+		return false, nil
+	}
+
+	// get uplink DR
+	uplinkDR, err := helpers.GetDataRateIndex(true, ctx.RXPacket.TXInfo, band.Band())
+	if err != nil {
+		return true, errors.Wrap(err, "get data-rate index error")
+	}
+
+	// get rx1 data-rate
+	drRX1Index, err := band.Band().GetRX1DataRateIndex(uplinkDR, int(ctx.DeviceSession.RX1DROffset))
+	if err != nil {
+		return true, errors.Wrap(err, "get rx1 data-rate index error")
+	}
+
+	if rx2OnlyDR >= 0 { // if the limit is set via config file, apply the settings accordingly
+		if rx2OnlyDR >= drRX1Index {
+			return true, nil
+		} else {
+			return false, nil
+		}
+	} else if rx2OnlyDR == -1 { // try to decide based on simplified link budget
+		var (
+			t			  int
+			txPowerRX1    int32
+			txPowerRX2    int32
+			freqRX1       uint32
+			freqRX2       uint32
+			linkBudgetRX1 float64
+			linkBudgetRX2 float64
+			drRX1		  loraband.DataRate
+			drRX2		  loraband.DataRate
+			logBWRX1 	  float64
+			logBWRX2	  float64
+		)
+
+		// get rx1 data-rate
+		drRX1, err = band.Band().GetDataRate(drRX1Index)
+		if err != nil {
+			return true, err // on error, omit RX1
+		}
+
+		if drRX1.Modulation == loraband.FSKModulation { //never block RX1 for FSK
+			return false, nil
+		}
+
+		// get rx2 data-rate
+		drRX2, err = band.Band().GetDataRate(int(ctx.DeviceSession.RX2DR))
+		if err != nil {
+			return false, err // on error, don't block RX1
+		}
+
+		// get RX1 and RX2 freq
+		t, err = band.Band().GetRX1FrequencyForUplinkFrequency(int(ctx.RXPacket.TXInfo.Frequency))
+		if err != nil {
+			return true, err //could not get RX1 Frequency for any reason, so just send RX2 then
+		} else {
+			freqRX1 = uint32(t)
+		}
+
+		freqRX2 = uint32(ctx.DeviceSession.RX2Frequency)
+
+		// get RX1 and RX2 TX Power
+		if downlinkTXPower != -1 {
+			txPowerRX1 = int32(downlinkTXPower)
+			txPowerRX2 = int32(downlinkTXPower)
+		} else {
+			txPowerRX1 = int32(band.Band().GetDownlinkTXPower(int(freqRX1)))
+			txPowerRX2 = int32(band.Band().GetDownlinkTXPower(int(freqRX2)))
+		}
+
+		if drRX1.Bandwidth == 125 {
+			logBWRX1 = 21 // 10*log10(125)
+		} else {
+			logBWRX1 = 24 // 10*log10(250)
+		}
+
+		if drRX2.Bandwidth == 125 {
+			logBWRX2 = 21 // 10*log10(125)
+		} else {
+			logBWRX2 = 24 // 10*log10(250)
+		}
+
+		// do the Link Budget calculation here (see http://www.techplayon.com/lora-link-budget-sensitivity-calculations-example-explained/)
+		// General formula: L = TX_Power - (-174 + 10 * log10(BW) + NF + minimal_SNR(SF))
+		// we assume a noise figure of 6dB, BW of 125 kHz (-> 10log(125) approx. 21) for simplicity
+		// TODO: introduce some more factors like asymmetric up- and down-paths due to local noise, other band widths, ...
+
+		linkBudgetRX1 = float64(txPowerRX1) - (float64(-174 + 6) + logBWRX1 + sfToSNR(drRX1.SpreadFactor))
+		linkBudgetRX2 = float64(txPowerRX2) - (float64(-174 + 6) + logBWRX2 + sfToSNR(drRX2.SpreadFactor))
+
+		if linkBudgetRX2 >= linkBudgetRX1 {
```

Comment:
Is there a reason to use RX2 when RX1 has the same budget? Doesn't have RX1 then the the preference as it is using multiple frequencies instead of just one fixed frequency?

Reply:
Actually, there is the factor of allowed air time/duty cycle per band. For example,  the 8 channels in EU868 (on 2 bands) have only 1% DC *per Band* (I think the 3 default channels are in one band, and the other 5 are in another one). This holds for uplink as well as for downlink (and therefore each gateway which has to server a lot of nodes simultaneously).

On the other hand, the 8692500 channel for downlink lays in a band with 10% DC, thereby having almost 6 times more allowed airtime per GW.

So a qualified decision had to take the current DC state of each gw into account. However, as we do not have these information, the better "guess" is to go with the band with the higher DC. 

And now comes the BUT: if I remember correctly, this information (1%, 10%) is currently not part of the system, right? So a optimization would include adding the DC for each band (or at least for some, and setting it to a default value for the others) and then consider the DC in addition, if the linkBudgets are identical.